stk/calculators/optimization/macromodel.py

```diff
@@ -1042,6 +1110,96 @@ def __init__(self,
                          minimum_gradient=minimum_gradient,
                          use_cache=use_cache)
 
+    @staticmethod
+    def _check_params(
+        temperature,
+        conformers,
+        simulation_time,
+        time_step,
+        eq_time,
+        minimum_gradient,
+        maximum_iterations
+    ):
+        """
+        Check if the optimization parameters are valid for MacroModel.
+
+        MacroModel ``.com`` files have well-defined column widths and
+        special ways to implement too long input. This method raises
+        `MacroModelInputError` exception if the input cannot be
+        converted into a valid ``.com`` file entry.
+
+        Parameters
+        ----------
+        temperature : :class:`float`, optional
+            The temperature in Kelvin at which the MD is run.
+
+        conformers' : :class:`int`, optional
+            The number of conformers sampled and optimized from the MD.
+
+        simulation_time : :class:`float`, optional
+            The simulation time in ``ps`` of the MD.
+
+        time_step : :class:`float`, optional
+            The time step in ``fs`` for the MD.
+
+        eq_time : :class:`float`, optional
+            The equilibriation time in ``ps`` before the MD is run.
+
+        minimum_gradient : :class:`float`, optional
+            The gradient at which optimization is stopped.
+
+        maximum_iterations : :class:`int`, optional
+            The maximum number of iterations done during the
+            optimization.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+
```

Comment:
I think the limits should be documented in the docstring, not in the code. In the code the limits are obvious, not so to the user.